## GROUP 1 PROJECT

**Author:** Itunuoluwa Isewon PhD      	
**Email:** itunu.isewon@covenantuniversity.edu.ng

## Phylogenetic Analysis of Fungal 28S Sequences

### Background

The 28S region of rDNA is widely used as a DNA barcode for fungi. In this project, students will analyse 28S sequences from five fungal species, along with a set of unknown fungal sequences. The aim is to use phylogenetic approaches to determine which species the unknown belongs to, and to compare different tree-building methods.

### Objectives

â€¢	Perform multiple sequence alignment (MSA) of fungal 28S sequences. 
â€¢	Construct phylogenetic trees using Neighbour Joining (NJ), Maximum Likelihood (ML), and Minimum Evolution (ME) methods.
â€¢	 Compare phylogenetic tree topologies across methods.
â€¢	 Identify the most likely species for the unknown sequences.

### Dataset Provided

### ðŸ“¥ **Dataset:**

FASTA file containing: 

â€¢	28S sequences from 5 known fungal species
â€¢	Several unknown 28S sequences to classify

Download dataset [here](https://drive.google.com/file/d/1Du-DAWhij6LZp5tCMSdW5byBG34W07t2/view?usp=sharing)

### Tasks

1.	Load the sequences Import the FASTA file into MEGA
2.	Multiple Sequence Alignment (MSA) Using ClustalW to align the 28S sequences.
3.	 Save the alignment file. 
4.	Phylogenetic Tree Construction: Build trees using three methods;
â€¢   Neighbor Joining (NJ)
â€¢	  Maximum Likelihood (ML)
â€¢  	Minimum Evolution (ME)
6.	Visualize each tree and annotate species names.
   
### Analysis

â€¢	Compare the positions of the unknown sequences in each tree. 
â€¢	Check if all methods agree on the classification of unknowns. 
â€¢	Note any differences in tree topology between methods.

### Expected Output
1.	A ClustalW alignment file
2.	Three phylogenetic trees (NJ, ML, ME)
3.	A presentation including: 
â€¢  	Figures of the phylogenetic trees 
â€¢	  Identification of unknown sequences
â€¢	  Discussion of agreement/disagreement among methods
